teps/0128-scheduled-runs.md

```diff
@@ -0,0 +1,393 @@
+For example:
+
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: ScheduledTrigger
+metadata:
+  name: nightly-release
+spec:
+  schedule: "0 0 * * *"
+  triggers:
+  - triggerRef: experimental-release-pipeline
+  serviceAccountName: default
+  cloudEventSink: http://eventlistener.free.beeceptor.com
+```
+
+The `schedule` field uses the same syntax as Kubernetes CronJobs.
+Like EventListeners, ScheduledTriggers can define Triggers or TriggerGroups inline, or contain references to Triggers.
+`serviceAccountName` defaults to "default", and `cloudEventSink` is optional.
+A ScheduledTrigger sends an event with an empty body.
+
+Consider a use case where a company has a release Pipeline they'd like to use for both nightly releases and event-driven releases.
+They can reuse a TriggerTemplate with this Pipeline in two different Triggers;
+one that is used in an EventListener and one that's used in a ScheduledTrigger.
+For example:
+
+```yaml
+kind: EventListener
+spec:
+  triggers:
+  - name: release
+    interceptors:
+    - ref:
+        name: github
+      params:
+      - name: "eventTypes"
+        value: ["push"]
+    bindings:
+    - name: git-sha
+      value: $(body.head_commit.id)
+    template:
+      ref: release-pipeline-template
+```
+
+```yaml
+kind: ScheduledTrigger
+spec:
+  schedule: "0 0 * * *"
+  triggers:
+  - bindings:
+    - name: git-sha
+      value: main
+    template:
+      ref: release-pipeline-template
+```
+
+Compared to the alternative solution of adding a [schedule to the Trigger CRD](#add-schedule-to-trigger),
+this solution has two benefits:
+1. Users can specify cloudEvent sinks for Triggers fired on a scheduled basis.
+2. Users can fire multiple Triggers on a scheduled basis with the same service account, using TriggerGroups.
+(For example, Tekton releases all of its experimental projects on a nightly basis with the same Pipeline.)
+
+### Bindings
+
+As part of future work for this proposal, we can add two new variable replacements in bindings:
+- $(context.date): date when the run was created, in RFC3339 format
+- $(context.datetime): timestamp when the run was created, in RFC3339 format
+
+Like EventListeners, ScheduledTriggers should generate an event ID (a UUID) and support variable replacement
+for $(context.eventID) in bindings.
+
+### CLI
+
+As part of future work for this proposal, we can add a CLI command to manually trigger execution of an EventListener or ScheduledTrigger
+similar to the kubectl functionality `kubectl create job --from=cronjob/my-cronjob`, for example:
+
+```sh
+tkn eventlistener start <el_name> --body "{'foo':'bar'}"
+```
+or
+
+```sh
+tkn crontrigger start <trigger_name>
+```
+
+This functionality is tracked in https://github.com/tektoncd/cli/issues/1833.
+
+## Design Evaluation
+
+### Reusability
+
+This feature is essentially syntactic sugar for a CronJob creating a Tekton resource.
+It reuses EventListener functionality such as TriggerGroups and CloudEventSinks.
```

Comment:
This only reuses EventListener functionality but it won't actually create an event listener, correct?

Reply:
yup! updated to clarify.